03-Make/cooking_soup/Makefile

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/make -f
+
+GOAL="Soup"
+
+POTATOES = potatoes
+CABBAGE = cabbage
+ONIONS = onions
+MEAT = meat
+CARROT = carrot
+SALT = salt
+
+QUANTITY_WATER="3 l"
+QUANTITY_POTATOES="800 g"
+QUANTITY_CABBAGE="1000 g"
+QUANTITY_ONIONS="350 g"
+QUANTITY_MEAT="500 g"
+QUANTITY_CARROT="200 g"
+QUANTITY_SALT="50 g"
+
+FOOD_MARKET = food_market
+KITCHEN = kitchen
+TABLE = table
+SPOON = spoon
+KNIFE = knife
+POT = pot
+SLICED_PRODUCTS = sliced_products
+DINNING_ROOM = dinning_room
+PRODUCTS = products
+
+
+TOTAL_GOAL_PRODUCTS = $(POTATOES).txt $(CABBAGE).txt \
+			$(ONIONS).txt $(MEAT).txt \
+			$(CARROT).txt $(SALT).txt
+
+CREATE_WITHOUT_MAKE = "create_without_make.sh"
+
+$(GOAL): create_directories $(TOTAL_GOAL_PRODUCTS) buy_products cut_products \
+	 move_sliced_products_to_pot bouling
+ifneq ($(ARG),-k)
+	@cat ./$(KITCHEN)/$(TABLE)/$(POT)/*.txt
+	@echo "$(GOAL) is cooked!"
+else
+	@echo "cat ./$(KITCHEN)/$(TABLE)/$(POT)/*.txt" >> $(CREATE_WITHOUT_MAKE)
+	@echo "echo '$(GOAL) is cooked!'" >> $(CREATE_WITHOUT_MAKE)
+endif
+create_directories:
+ifneq ($(ARG),-k)
+	echo "ARG=$(ARG)"
+	mkdir -vp ./$(FOOD_MARKET)/
+	mkdir -vp ./$(KITCHEN)/$(TABLE)/$(SPOON)
+	mkdir -vp ./$(KITCHEN)/$(TABLE)/$(KNIFE)
+	mkdir -vp ./$(KITCHEN)/$(TABLE)/$(POT)
+	mkdir -vp ./$(KITCHEN)/$(PRODUCTS)
+	mkdir -vp ./$(KITCHEN)/$(SLICED_PRODUCTS)
+	mkdir -vp ./$(DINNING_ROOM)
+	touch ./$(KITCHEN)/water.txt
+else
+	touch ./$(CREATE_WITHOUT_MAKE)
```

Comment:
What is the reason for this touch in cases of writing to file in the followed line. May be running chmod after stdout redirection is enough?